XSConsoleAuth.py

```diff
@@ -109,26 +106,7 @@ def TCPAuthenticate(self, inUsername, inPassword):
                 session.close()
 
     def PAMAuthenticate(self, inUsername, inPassword):
-
-        def PAMConv(inAuth, inQueryList, *theRest):
-            # *theRest consumes the userData argument from later versions of PyPAM
-            retVal = []
-            for query in inQueryList:
-                if query[1] == PAM.PAM_PROMPT_ECHO_ON or query[1] == PAM.PAM_PROMPT_ECHO_OFF:
-                    # Return inPassword from the scope that encloses this function
-                    retVal.append((inPassword, 0)) # Append a tuple with two values (so double brackets)
-            return retVal
-
-        auth = PAM.pam()
-        auth.start('passwd')
-        auth.set_item(PAM.PAM_USER, inUsername)
-        auth.set_item(PAM.PAM_CONV, PAMConv)
-
-        try:
-            auth.authenticate()
-            auth.acct_mgmt()
-            # No exception implies a successful login
-        except Exception, e:
+        if not pam.authenticate(inUsername, inPassword):
```

Comment:
This fixes PAM Authentication (for sure on XS8, please test for XS9 too):

```suggestion
        if not pam.authenticate(inUsername, inPassword, service="passwd"):
```
It was really easy to find using Googling '"pam.authentication" False python':

The default argument is: `pam.authenticate(user, pass, service="login")`

`service="login"` also checks if the user's tty is in `/etc/securetty`

On XS8, and (among serial lines, etc) there is only `pts/0` in `/etc/securetty`.

But if you are logged in over SSH, you might get the pty `pts/2`, for example.

When `pam.authenticate(user, pass, service="login")` does not find your SSH pty in `/etc/securetty`, it rejects the login.

Thus, it may happen if you open multiple SSH sessions and also test it on one of the sessions which don't have `pts/0`, it could fail on XS9 too:

Example command line:
```py
python3 -c 'import pam;print(pam.authenticate("root","root", service="login"))'
False
```

But for us, for compatibility, just checking the password using `service="passwd") is sufficient (like the old python2 PAM auth does).

It is possible that the same happens on xs9 too, but you might not have run into it if you only tested xsconsole on the default terminals and never called it on a second or third SSH pty.

Reference: https://github.com/FirefighterBlu3/python-pam/issues/13